R/loadFromMothur.R

```diff
@@ -106,27 +115,30 @@ loadFromMothur <- function(featureTableFile,
         taxa_tab <- S4Vectors:::make_zero_col_DataFrame(nrow(feature_tab))
     }
     
-    # If colData information exists, gets that. Otherwise, sample_tab is just data frame without information
-    if (!is.null(sampleMetaFile)) {
-        sample_meta <- .read_mothur_sample_meta(sampleMetaFile, feature_tab)
+    # If colData informationor data_to_colData exists, gets that. Otherwise, sample_tab is just data frame without information
+    if (!is.null(sampleMetaFile) %% !is.null(data_to_colData)) {
+        sample_meta <- .read_mothur_sample_meta(sampleMetaFile, data_to_colData)
     } else {
         sample_meta <- S4Vectors:::make_zero_col_DataFrame(ncol(feature_tab))
     }
     
-    if (!is.null(phyTreeFile)) {
-        tree <- .read_qza(phyTreeFile, ...)
-    } else {
-        tree <- NULL
-    }
+    tree <- NULL
+    refseq <- NULL
     
-    # if row.names(feature_tab) is a DNA sequence,  set it as refseq
-    if (!is.null(refSeqFile)){
-        refseq <- .read_qza(refSeqFile, ...)
-    } else if (featureNamesAsRefSeq) {
-        refseq <- .rownames_as_dna_seq(rownames(feature_tab))
-    } else {
-        refseq <- NULL
-    }
+    # if (!is.null(phyTreeFile)) {
+    #     tree <- .read_qza(phyTreeFile, ...)
+    # } else {
+    #     tree <- NULL
+    # }
+    # 
+    # # if row.names(feature_tab) is a DNA sequence,  set it as refseq
+    # if (!is.null(refSeqFile)){
+    #     refseq <- .read_qza(refSeqFile, ...)
+    # } else if (featureNamesAsRefSeq) {
+    #     refseq <- .rownames_as_dna_seq(rownames(feature_tab))
+    # } else {
+    #     refseq <- NULL
+    # }
     
     return(TreeSummarizedExperiment(
```

Comment:
Since the input from mothur, cannot require `TSE` to be used (now reference sequence, no tree), I would recommend just returning a `SE`.

Reply:
tse --> se